files/chaincode/src/map/map.go

```diff
@@ -41,27 +42,52 @@ import (
 type SimpleChaincode struct {
 }
 
+func ensureStringIsUTF8Valid(inputString string ) string {
```

Comment:
it's better not to skip invalid characters, but check if there any invalid and stop processing with error

Reply:
This function used in following context:

```
// Init is a no-op
func (t *SimpleChaincode) Init(stub shim.ChaincodeStubInterface) pb.Response {
	creator, err := stub.GetCreator();
	creatorString := ensureStringIsUTF8Valid(string(creator[:]))
```

and it its turn

```
  // GetCreator returns `SignatureHeader.Creator` (e.g. an identity)
  // of the `SignedProposal`. This is the identity of the agent (or user)
  // submitting the transaction.
  GetCreator() ([]byte, error)
```

And this SignedProposal is built by Fabric SDK, so we have no control over this logic and hence it's not possible to fix the test.

I believe this conversion to UTF8 can be considered as normalisation of data on chaincode side

Reply:
I still believe it's better to transform byte string to valid string completely, without throwing some bytes away, due to two reasons:
1) there is not only X509Identity in fabric, there are IdeMix identities as well, which is much more complex and have lots of byte strings embedded, so loss might be non-negligiable
2) This behaviour might have changed in the future versions of Fabric (f.e., they can attach binary version of certificate eventually), but we would miss it silently

Thus I propose to use something like url.PathEscape/url.PathUnescape to convert to valid string with not much overhead

Reply:
Fixed